### PR TITLE
doc: update the ntp configuration

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
Add the document for the NTP configuration.

### CODE REVIEW

File: docs/host/host.md [hunk 1]
```diff
@@ -251,3 +251,32 @@ When Ksmtuned starts, initialize `pages_to_scan` in KSM to 1000 (**Minimum Pages
...
+
+In v1.2.0, Harvester supports NTP configuration on Harvester/Settings. You can configure NTP servers anytime for the whole Harvester cluster. The NTP configuration will be applied to all nodes in the cluster.
+
+![](/img/v1.3/host/harvester-ntp-settings.png)
+
+You could setup multiple NTP Servers like the following figure.
+
+![](/img/v1.3/host/harvester-ntp-settings-multiple.png)
+
+You can check the configuration result with CRD `node.harvesterhci.io/ntp-service`
```
Comment (bk201) [inline docs/host/host.md:269]:
```suggestion
You can check the configuration result from the `node.harvesterhci.io/ntp-service` annotation in Kubernetes Nodes: 
```

File: docs/host/host.md [hunk 2]
```diff
@@ -251,3 +251,32 @@ When Ksmtuned starts, initialize `pages_to_scan` in KSM to 1000 (**Minimum Pages
...
+- `ntpSyncStatus`: synced or unsynced
+- `currentNtpServers`: current NTP servers
+
+```
+$ kubectl get nodes harvester-node-0 -o yaml |yq -e '.metadata.annotations.["node.harvesterhci.io/ntp-service"]'
+{"ntpSyncStatus":"synced","currentNtpServers":"0.suse.pool.ntp.org 1.suse.pool.ntp.org"}
+```
+
+There are points you need to know: 
+1. Do not directly modify the NTP config on the host. Node-manager will keep the consistency between the configuration on Harvester/Settings and the configuration on the host.
```
Comment (bk201) [inline docs/host/host.md:279]:
```suggestion
1. Do not modify the NTP config on hosts. Harvester will keep syncing the configuration from Harvester/Settings to hosts.
```

File: docs/host/host.md [hunk 3]
```diff
@@ -251,3 +251,32 @@ When Ksmtuned starts, initialize `pages_to_scan` in KSM to 1000 (**Minimum Pages
...
+2. Suppose your system is upgraded from the previous version. In that case, you need to manually configure your NTP servers on Harvester/Settings first because Harvester does not know your last NTP configuration if there are conflicts. So, for the upgrade case, your ntp-server will be empty, like the following figure.
```
Comment (bk201) [inline docs/host/host.md:280]:
```suggestion
2. Suppose your system is upgraded from a previous version. In that case, you need to manually configure your NTP servers on Harvester/Settings first because Harvester does not know your last NTP configuration if there are conflicts. So, for the upgrade case, your ntp-server will be empty, like the following figure.
```

File: docs/install/update-harvester-configuration.md [hunk 4]
```diff
@@ -71,86 +71,7 @@ If you upgrade from a version before `v1.1.2`, the `cloud-init` file in examples
...
-    stages:
-      network:
-      - ...
-        authorized_keys:
-          rancher:
-          - key1
-          - key2
-    ```
-
+After v1.2.0 and later versions, if you want to change the NTP configuration runtime, please refer to the [NTP servers](../host/host.md#ntp-configuration) section.
```
Comment (bk201) [inline docs/install/update-harvester-configuration.md:74]:
```suggestion
After v1.2.0 and later versions, if you want to change the NTP configuration at runtime, please refer to the [NTP servers](../host/host.md#ntp-configuration) section.
```

@jillian-maroket please help check the preposition here, thanks.

File: docs/host/host.md [hunk 5]
```diff
@@ -251,3 +251,32 @@ When Ksmtuned starts, initialize `pages_to_scan` in KSM to 1000 (**Minimum Pages
 KSM starts when the available memory falls below the **Threshold Coefficient**. If it detects that it is running, `pages_to_scan` increments by 300 (**Boost**) every minute until it reaches 5000 (**Maximum Pages**).
 
 KSM will stop when the available memory is above the **Threshold Coefficient**. If it detects that it is stopped, `pages_to_scan` decrements by 100 (**Decay**) every minute until it reaches 1000 (**Minimum Pages**).
+
+## NTP Configuration
+
+Time synchronization is important for the distributed cluster architecture, so there is a simple way to configure the NTP servers on Harvester.
+
+In the previous version, we only support NTP configuration when [installation](https://docs.harvesterhci.io/v1.2/install/harvester-configuration#osntp_servers). If you want to change the NTP configuration, manually update the configuration file on each node.
```
Comment (connorkuehl) [inline docs/host/host.md:259]:
```suggestion
In the previous version, we only support NTP configuration [during the installation process](https://docs.harvesterhci.io/v1.2/install/harvester-configuration#osntp_servers). If you want to change the NTP configuration, it was necessary to manually update the configuration file on each node.
```

File: docs/host/host.md [hunk 6]
```diff
@@ -251,3 +251,32 @@ When Ksmtuned starts, initialize `pages_to_scan` in KSM to 1000 (**Minimum Pages
...
+
+In v1.2.0, Harvester supports NTP configuration on Harvester/Settings. You can configure NTP servers anytime for the whole Harvester cluster. The NTP configuration will be applied to all nodes in the cluster.
```
Comment (connorkuehl) [inline docs/host/host.md:261]:
```suggestion
In v1.2.0, Harvester supports NTP configuration in Harvester/Settings. You can configure NTP servers any time for the whole Harvester cluster. The NTP configuration will be applied to all nodes in the cluster.
```

File: docs/host/host.md [hunk 7]
```diff
@@ -251,3 +251,32 @@ When Ksmtuned starts, initialize `pages_to_scan` in KSM to 1000 (**Minimum Pages
...
+
+![](/img/v1.3/host/harvester-ntp-settings.png)
+
+You could setup multiple NTP Servers like the following figure.
```
Comment (connorkuehl) [inline docs/host/host.md:265]:
```suggestion
You could setup multiple NTP Servers like the following figure:
```

**Review by connorkuehl — APPROVED** [review]
Just some minor phrasing suggestions, otherwise the content looks good to me.

File: docs/host/host.md [hunk 8]
```diff
@@ -251,3 +251,32 @@ When Ksmtuned starts, initialize `pages_to_scan` in KSM to 1000 (**Minimum Pages
 KSM starts when the available memory falls below the **Threshold Coefficient**. If it detects that it is running, `pages_to_scan` increments by 300 (**Boost**) every minute until it reaches 5000 (**Maximum Pages**).
 
 KSM will stop when the available memory is above the **Threshold Coefficient**. If it detects that it is stopped, `pages_to_scan` decrements by 100 (**Decay**) every minute until it reaches 1000 (**Minimum Pages**).
+
+## NTP Configuration
+
+Time synchronization is important for the distributed cluster architecture, so there is a simple way to configure the NTP servers on Harvester.
```
Comment (jillian-maroket) [inline docs/host/host.md:257]:
```suggestion
Time synchronization is an important aspect of distributed cluster architecture. Because of this, Harvester now provides a simpler way for configuring NTP settings.
```

File: docs/host/host.md [hunk 9]
```diff
@@ -251,3 +251,32 @@ When Ksmtuned starts, initialize `pages_to_scan` in KSM to 1000 (**Minimum Pages
...
+
+In the previous version, Harvester only supported NTP configuration [during the installation process](https://docs.harvesterhci.io/v1.2/install/harvester-configuration#osntp_servers). If you want to change the NTP configuration, it was necessary to manually update the configuration file on each node.
```
Comment (jillian-maroket) [inline docs/host/host.md:259]:
```suggestion
In previous Harvester versions, NTP settings were mainly configurable [during the installation process](https://docs.harvesterhci.io/v1.2/install/harvester-configuration#osntp_servers). To modify the settings, you needed to manually update the configuration file on each node.
```

File: docs/host/host.md [hunk 10]
```diff
@@ -251,3 +251,32 @@ When Ksmtuned starts, initialize `pages_to_scan` in KSM to 1000 (**Minimum Pages
...
+
+In v1.2.0, Harvester supports NTP configuration in Harvester/Settings. You can configure NTP servers any time for the whole Harvester cluster. The NTP configuration will be applied to all nodes in the cluster.
```
Comment (jillian-maroket) [inline docs/host/host.md:261]:
```suggestion
Beginning with version v1.2.0, Harvester is supporting NTP configuration on the Harvester UI Settings screen (**Advanced** > **Settings**). You can configure NTP settings for the entire Harvester cluster at any time, and the settings are applied to all nodes in the cluster.
```

File: docs/host/host.md [hunk 11]
```diff
@@ -251,3 +251,32 @@ When Ksmtuned starts, initialize `pages_to_scan` in KSM to 1000 (**Minimum Pages
...
+
+![](/img/v1.3/host/harvester-ntp-settings.png)
+
+You could setup multiple NTP Servers like the following figure:
```
Comment (jillian-maroket) [inline docs/host/host.md:265]:
```suggestion
You can set up multiple NTP servers at once.
```

File: docs/host/host.md [hunk 12]
```diff
@@ -251,3 +251,32 @@ When Ksmtuned starts, initialize `pages_to_scan` in KSM to 1000 (**Minimum Pages
...
+
+![](/img/v1.3/host/harvester-ntp-settings-multiple.png)
+
+You can check the configuration result from the `node.harvesterhci.io/ntp-service` annotation in Kubernetes Nodes: 
+- `ntpSyncStatus`: synced or unsynced
+- `currentNtpServers`: current NTP servers
```
Comment (jillian-maroket) [inline docs/host/host.md:271]:
```suggestion
You can check the settings in the `node.harvesterhci.io/ntp-service` annotation in Kubernetes nodes: 
- `ntpSyncStatus`: Status of the connection to NTP servers (possible values: `synced` and `unsynced`)
- `currentNtpServers`: List of existing NTP servers
```

File: docs/host/host.md [hunk 13]
```diff
@@ -251,3 +251,32 @@ When Ksmtuned starts, initialize `pages_to_scan` in KSM to 1000 (**Minimum Pages
...
+
+```
+$ kubectl get nodes harvester-node-0 -o yaml |yq -e '.metadata.annotations.["node.harvesterhci.io/ntp-service"]'
+{"ntpSyncStatus":"synced","currentNtpServers":"0.suse.pool.ntp.org 1.suse.pool.ntp.org"}
+```
+
+There are points you need to know: 
+1. Do not modify the NTP config on hosts. Harvester will keep syncing the configuration from Harvester/Settings to hosts.
+2. Suppose your system is upgraded from a previous version. In that case, you need to manually configure your NTP servers on Harvester/Settings first because Harvester does not know your last NTP configuration if there are conflicts. So, for the upgrade case, your ntp-server will be empty, like the following figure.
```
Comment (jillian-maroket) [inline docs/host/host.md:280]:
```suggestion
> **Note:**
>
> 1. Do not modify the configuration file on each node. Harvester will automatically sync the settings that you configured on the Harvester UI to the nodes.
> 1. If you upgraded Harvester from an earlier version, the **ntp-servers** list on the Settings screen will be empty (see screenshot). You must manually configure the NTP settings because Harvester is unaware of the previous settings and is unable to detect conflicts.
```

File: docs/host/host.md (hunk 8 above)
Comment (jillian-maroket) [reply docs/host/host.md:257]: Because the Harvester UI uses the term "Settings", we should make it easy for users to map the concepts to that particular screen. Here is the wording convention that I used:
- "configure" and "configuration": refer to the action the user performs
- "setting": refers to what the user configures

File: docs/host/host.md (hunk 10 above)
Comment (jillian-maroket) [reply docs/host/host.md:261]: The phrase "configure NTP servers" implies that the user is configuring the actual server, which is not the case here. "NTP settings" covers server addresses and other settings related to the specified servers.

File: docs/host/host.md (hunk 12 above)
Comment (jillian-maroket) [reply docs/host/host.md:271]: Please check if my understanding is correct. I could not find information on the annotation mentioned here.

File: docs/host/host.md (hunk 13 above)
Comment (jillian-maroket) [reply docs/host/host.md:280]: I converted this block of text to a note so that it stands out. I followed the markup described here: https://www.markdownguide.org/hacks/#admonitions

File: docs/install/update-harvester-configuration.md (hunk 4 above)
Comment (jillian-maroket) [reply docs/install/update-harvester-configuration.md:74]: Both "NTP configuration at runtime" and "NTP runtime configuration" can make sense in this context, and I don't know which one you intended to say. 

In any case, I would reword the sentence to the following:
For more information about...INSERT PHRASE...in Harvester 1.2.0 and later versions, see [NTP servers](../host/host.md#ntp-configuration).

**Review by jillian-maroket — COMMENTED** [review]
@Vicente-Cheng I explained the terms and wording choices in the comments. Please check.

File: docs/install/update-harvester-configuration.md (hunk 4 above)
Comment (Vicente-Cheng) [reply docs/install/update-harvester-configuration.md]: I also update this section. Please help to recheck it!

**Review by jillian-maroket — APPROVED** [review]
LGTM. Thanks!

**Comment by bk201** [comment]
lgtm, we can merge it after adding v1.2 contents

**Comment by Vicente-Cheng** [comment]
1.2 content added. @bk201, please help to recheck it. Thanks!

**Review by bk201 — APPROVED** [review]
